### DISCUSSION TITLE
Pretty-printing Jsonnet expressions

### DISCUSSION THREAD

**Discussion by moleike** [moleike/haskell-jsonnet#51] [General]
@ozkutuk adding this here in case you want to continue the discussion: https://github.com/moleike/haskell-jsonnet/pull/48#issuecomment-1020922037

> Well, if we did this from the ground up, I would probably prefer using aeson-pretty

I didn't know about that library at the time, but also I don't think it's a bad idea to have our own JSON pretty-printer and avoid yet another dependency.

> We don't do any codegen stuff, so is it just for the sake of having a roundtrip test?

So that was me working towards adding gradual typing to jsonnet. I was looking into adding this as a separate option, that would type check jsonnet with type annotations and output vanilla jsonnet with typing information discarded. I figured I needed to be able to output jsonnet, so that's why I did that (in fact that's the last thing I did before stop making progress). But you are right, that code is pretty useless at the moment, and we could even potentially remove it.

Other things in the pretty print module are for pretty-printing errors, which again need a `Pretty` instance, and I decided to put everything in its own module rather than having the instances scattered around.

HIH

**Comment by ozkutuk** [comment]
I see, I would love to see gradual typing in haskell-jsonnet, even potentially working on it! In that case, I think it can remain in code for the time being. Though I see imports like this in various different modules:
```hs
import Language.Jsonnet.Pretty ()
```
I think we can remove these as it adds to the confusion. As long as it is obvious that prettyprinting is not really used for the time being, we are good.

**Reply by moleike** [reply]
We have instances of many other types, besides Jsonnet expressions. If you think it can be removed, please go ahead and remove it! 👍
